sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/interfaces.scala

```diff
@@ -187,7 +184,7 @@ sealed abstract class AggregateFunction extends Expression with ImplicitCastInpu
  * `inputAggBufferOffset`, but not on the correctness of the attribute ids in `aggBufferAttributes`
  * and `inputAggBufferAttributes`.
  */
-abstract class ImperativeAggregate extends AggregateFunction {
+abstract class ImperativeAggregate extends AggregateFunction with CodegenFallback {
```

Comment:
We should be fine as long as we add already bound ```ImperativeAggregate```s to the projection to be code generated. Unbound ```ImperativeAggregate```s will cause alot of trouble.

I use ```HyperLogLogPlusPlus``` in the last test in the DataFrameWindowFunctionSuite: https://github.com/hvanhovell/spark/blob/SPARK-8641-2/sql/core/src/test/scala/org/apache/spark/sql/DataFrameWindowSuite.scala#L222 is this enough?